packages/mobile/src/navigator/QRNavigator.tsx

```diff
@@ -99,13 +99,16 @@ function AnimatedScannerScene({ route, position, ...props }: AnimatedScannerScen
   const onBarCodeDetected = memoize(
     (qrCode: QrCode) => {
       Logger.debug('QRScanner', 'Bar code detected')
+
+      const isWalletConnectRequest = qrCode.data.startsWith('wc:') || undefined
```

Comment:
Nit: maybe instead of a new param for handleBarcodeDetected, just put this logic in the the saga?